templates/tfengine/components/devops/variables.tf

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+{{if not (get .admins_group "exists" false)}}
```

Comment:
Nit: This is nit-picking, but I'd find it easier to read if phrased in the positive:

```
{{$must_create_admins_group := not (get .admins_group "exists" false)}}
{{if $must_create_admins_group}}
...
```

Reply:
If this variable is only used once, maybe just add comment above should be enough?

Reply:
Agree that we can rephrase this to be a positive expression, but it's the only case for this variable as @xingao267 said. Also, I found this case of a single-used negated variable is common. Whether we decide to work with at most 1 `not` and let these cases pass, or decide to use intermediate variables such as `$must_create_admins_group` and we will carry some technical debt for other expressions left in the project that are the same.

Personally I think this is pretty clear if we just remove the `false` default. Otherwise it will return `nil`, which is also falsy.

For now I put this line as follows:

```
{{if not (get .admins_group "exists")}}
```

But I'll keep this open so we can discuss further

Reply:
If `nil` if falsey, I'm actually fine with just the "if not exists", it's a small difference but it's sufficiently readable IMO.